core/internal/clientstate/client-state.go

```diff
@@ -156,10 +128,8 @@ func (c *clientState) AddReply(reply *messages.Reply) error {
 	c.Lock()
 	defer c.Unlock()
 
-	if seq != c.lastAcceptedSeq {
-		return fmt.Errorf("request ID mismatch")
-	} else if seq == c.lastExecutedSeq {
-		return fmt.Errorf("duplicate Reply")
+	if seq <= c.lastExecutedSeq {
```

Comment:
Currently request seq number is generated based on timestamp. So if an attacker creates a request message on the system with wrong clock and large sequence number is used, any messages generated based on proper timestamp could be discarded as old requests?

Reply:
Please note that request identifiers (seq) are independent for each client. So if a compromised client exhausts its request ID space, it is screwed up. But only this client identity would be affected; others could continue normally. Although the consensus protocol cannot protect the client from being compromised, it provides safety and liveness when up to f replica nodes may be compromised (except tamperproof component).

Reply:
OK, thank you for pointing out.